common/src/main/java/org/dash/wallet/common/Configuration.java

```diff
@@ -437,10 +437,17 @@ public void setShowJoinDashPay(boolean showJoinDashPay) {
     }
 
     public long getLastSeenNotificationTime() {
-        return prefs.getLong(PREFS_LAST_SEEN_NOTIFICATION_TIME, 0);
+        long savedTime = prefs.getLong(PREFS_LAST_SEEN_NOTIFICATION_TIME, 0);
+        long distantFutureTime = 2524608000L; //2050-01-01
+        if (savedTime > distantFutureTime) {
+            // date incorrectly saved in microseconds, fix it
+            savedTime = savedTime / 1000;
+            setPrefsLastSeenNotificationTime(savedTime);
+        }
+        return savedTime;
     }
 
     public void setPrefsLastSeenNotificationTime(long lastSeenNotificationTime) {
-        prefs.edit().putLong(PREFS_LAST_SEEN_NOTIFICATION_TIME, lastSeenNotificationTime).commit();
+        prefs.edit().putLong(PREFS_LAST_SEEN_NOTIFICATION_TIME, lastSeenNotificationTime).apply();
```

Comment:
this was changed from .apply() to .commit() to ensure that the setting would actually be saved right away.

Otherwise a user can leave the Notifications screen which has New items and then return immediately and the New items are still new, rather than Earlier.

Is there a better way?  Does/can this cause an ANR?

Reply:
I'm not sure if there is a need for protect against mentioned issue.
The documentation of `apply()` method reads:
```As SharedPreferences instances are singletons within a process, it's safe to replace any instance of commit() with apply() if you were already ignoring the return value.```
And we actually ignore the return value.